sql/dd/impl/upgrade/server.cc

```diff
@@ -518,6 +518,48 @@ bool upgrade_help_tables(THD *thd) {
   return false;
 }
 
+/*
+  For SYSTEM table, Run alter tables with target ddse during dd engine change
+*/
+bool fix_system_tables_for_engine_change(THD *thd) {
+  if (dd::execute_query(thd, "USE mysql")) {
+    LogErr(ERROR_LEVEL, ER_DD_UPGRADE_FAILED_FIND_VALID_DATA_DIR);
+    assert(false);
+    return true;
+  }
+
+  // iterate SYSTEM tables and run alter engine=target ddse
+  for (System_tables::Const_iterator it =
+           System_tables::instance()->begin(System_tables::Types::SYSTEM);
+       it != System_tables::instance()->end();
+       it = System_tables::instance()->next(it, System_tables::Types::SYSTEM)) {
+    assert((*it)->property() == System_tables::Types::SYSTEM);
+
+    dd::String_type table_name = (*it)->key().first + "." + (*it)->key().second;
+
+    if (table_name == "mysql.gtid_executed" ||
```

Comment:
Need a comment here. It might make sense to store these names in some structure or array, depending on its size